src/core/Service.ts

```diff
@@ -4,18 +4,23 @@
  */
 
 import { EventEmitter } from "eventemitter3";
+import {
+  RosbridgeCallServiceMessage,
+  RosbridgeServiceResponseMessage,
+} from "../types/protocol.ts";
+import Ros from "./Ros.js";
 
 /**
  * A ROS service client.
- * @template TRequest, TResponse
  */
-export default class Service extends EventEmitter {
+export default class Service<TRequest, TResponse> extends EventEmitter {
   /**
    * Stores a reference to the most recent service callback advertised so it can be removed from the EventEmitter during un-advertisement
    * @private
-   * @type {((rosbridgeRequest) => any) | null}
    */
-  _serviceCallback = null;
+  _serviceCallback:
+    | ((rosbridgeRequest: RosbridgeCallServiceMessage<TRequest>) => void)
```

Comment:
Shouldn't this return `void | Promise<void>` since in advertiseAsync you assign an async function?